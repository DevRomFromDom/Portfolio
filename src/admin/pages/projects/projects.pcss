.page-content {
    padding: 60px 36px 0 60px;
    background: url("../../../images/content/admin.jpg") center center / cover
        no-repeat;
    flex: 1;
    @media screen and (max-width: 768px) {
        padding: 40px 0 0 0;

    }
    @media screen and (max-width: 480px) {
        padding: 35px 0px 0px 0px;
    }
}
.header {
    display: flex;
    margin-bottom: 52px;
    @media screen and (max-width: 480px) {
        flex-direction: column;
        margin-left: 20px;
        justify-content: space-between;
        line-height: 25px;
        font-size: 18px;
        margin-bottom: 40px;
        height: 60px;
    }
}
.title {
    margin-right: 60px;
    font-size: 21px;
    font-weight: bold;
    color: $text-color;
    @media screen and (max-width: 768px) {
        margin-left: 30px;
    }
    @media screen and (max-width: 480px ) {
      font-size: 18px;
      line-height: 25px;
      
    }
  }
.projects__content-container{
    width: 100%;
    display: flex;
    flex-wrap: wrap;
    @media screen and (max-width: 768px ) {
        padding: 0 28px
         
       }
   ;
    @media screen and (max-width: 480px) {
        padding: 0;
    }
}
.create-project{
    display: flex;
    width: calc(100%/3 - 28px);
    height: 580px;
    max-height: 580px;
    margin-right: 28px;
    margin-bottom: 28px;
    &:hover{
        opacity: 0.8;
    }
    @media screen and (max-width: 768px ) {
       width: calc(100%/2-28px);
       
        
      }

      @media screen and (max-width: 480px) {
        width: 100%;
        margin: 0 auto;
        height: auto;
        max-width: 340PX;
    }
}