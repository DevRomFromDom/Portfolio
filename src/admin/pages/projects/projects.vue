<template>
    <div class="page-content">
        <div class="header">
            <div class="title">Блок «Работы»</div>
        </div>
        <div class="projects__content-container">
            <create-new-project
                class="cnp"
                v-if="projectModalIsAcrive"
                :modalTitle="modalTitle"
                @closeModal="closeModal"
                :editData="editProjectData" 
                @reset="resetModal"
            />
            <div class="create-project" v-if="projectModalIsAcrive === false">
                <square-btn
                    class="create-button"
                    type="square"
                    title="Добавить работу"
                    @click="addNewProject"
                />
            </div> 
            <project-cart
                v-for="item in projects"
                :key="item.id"
                :project="item"
                @remove="removeProject($event)"
                @editProject="editProject($event)"
                :modalOpen="projectModalIsAcrive"
                :editCurProject="editProjectData.id === item.id"
            />
        </div>
    </div>
</template>

<script>
import projectCart from "../../components/projectCart";
import squareBtn from "../../components/button";
import CreateNewProject from "../../components/createNewProject/createNewProject.vue";
import { mapActions, mapState } from "vuex";
export default {
   
    computed: {
        ...mapState("projects", { projects: (state) => state.data }),
    },
    data() {
        return {
            projectModalIsAcrive: false,
            modalTitle: "Добавление работы",
            editProjectData: {},
        };
    },
    components: {
        projectCart,
        squareBtn,
        CreateNewProject,
    },
    methods: {
        ...mapActions({
            getProjectsAction: "projects/getAllProjects",
            removeProjectAction: "projects/deleteProject",
        }),
        closeModal() {
            this.projectModalIsAcrive = false;
        },
        addNewProject() {
            this.modalTitle = "Добавление работы";
            this.projectModalIsAcrive = true;
        },
        removeProject(id) {
            if (this.projectModalIsAcrive !== false) return;
            this.removeProjectAction(id);
        },
        editProject(project) {
            if (this.projectModalIsAcrive !== false) return;
            const editProj = project;
            this.editProjectData = editProj;
            this.modalTitle = "Редактирование работы";
            this.projectModalIsAcrive = true;
        },
        resetModal() {
            this.editProjectData = {};
            this.projectModalIsAcrive = false;
        },
    },
    created() {
        this.getProjectsAction();
    },
};
</script>

<style lang="postcss" scoped src="./projects.pcss"></style>
