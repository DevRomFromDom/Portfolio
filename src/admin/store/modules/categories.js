export default {
    namespaced: true,
    state: {
        data: [],
    },
    mutations: {
        SET_CATEGORIES: (state, categories) => (state.data = categories),
        ADD_CATEGORY: (state, category) => state.data.unshift(category),
        DELETE_CATEGORY: (state, id) =>
            (state.data = state.data.filter((el) => el.id !== id)),
        EDIT_CATEGORY: (state, { title, id }) =>
            (state.data = state.data.map((cat) => {
                if (cat.id === id) {
                    return { ...cat, title: title };
                }
                return cat;
            })),
        ADD_SKILL: (state, data) =>
            (state.data = state.data.map((el) => {
                if (el.id === data.category) {
                    el.skills.push(data);
                }
                return el;
            })),
        DELETE_SKILL: (state, { skillId, categoryId }) =>
            (state.data = state.data.map((category) => {
                if (category.id === categoryId) {
                    return {
                        ...category,
                        skills: category.skills.filter(
                            (skill) => skill.id !== skillId
                        ),
                    };
                }
                return category;
            })),
        EDIT_SKILL: (state, skill) =>
            (state.data = state.data.map((category) => {
                if (category.id === skill.category) {
                    return {
                        ...category,
                        skills: category.skills.map((item) => {
                            if (item.id === skill.id) {
                                return {
                                    ...item,
                                    title: skill.title,
                                    percent: skill.percent,
                                };
                            }
                            return item;
                        }),
                    };
                }
                return category;
            })),
    },
    actions: {
        async getCategories({ commit,dispatch }) {
            try {
                const userData = await this.$axios.get("/user");
                const {data} = await this.$axios.get(
                    `/categories/${userData.data.user.id}`
                );
                commit("SET_CATEGORIES", data);
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async createCategory({ dispatch, commit }, title) {
            try {
                const { data } = await this.$axios.post("/categories", {
                    title,
                });
                commit("ADD_CATEGORY", data);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Добавлена новая категория",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async removeCategory({ dispatch, commit }, id) {
            try {
                const res = await this.$axios.delete(`/categories/${id}`);
                console.log(res)
                commit("DELETE_CATEGORY", id);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Категория удалена",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async editCatigoryTitle({ dispatch, commit }, event) {
            try {
                const res = await this.$axios.post(
                    `/categories/${event.categoryId}`,
                    event
                );
                commit("EDIT_CATEGORY", { event });
                dispatch(
                    "tooltips/show",
                    {
                        text: "Название категории изменено",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async addSkill({ dispatch, commit }, skill) {
            try {
                const { data } = await this.$axios.post("/skills", skill);
                commit("ADD_SKILL", data);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Добавлено новое умение",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async removeSkill({ dispatch, commit }, { skillId, categoryId }) {
            try {
                const res = await this.$axios.delete(`/skills/${skillId}`);
                commit("DELETE_SKILL", { skillId, categoryId });
                dispatch(
                    "tooltips/show",
                    {
                        text: "Умение удалено",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async editSkill({ dispatch, commit }, skill) {
            try {
                const res = await this.$axios.post(`/skills/${skill.id}`, {
                    title: skill.title,
                    percent: Number(skill.percent),
                    category: skill.category,
                });
                commit("EDIT_SKILL", skill);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Значения умения обновлены",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
    },
};
