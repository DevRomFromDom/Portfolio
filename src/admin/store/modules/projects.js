export default {
    namespaced: true,
    state: {
        data: [],
    },
    mutations: {
        ADD_PROJECT: (state, data) => state.data.push(data),
        GET_PROJECTS: (state, data) => (state.data = data),
        DELETE_PROJECT: (state, id) =>
            (state.data = state.data.filter((el) => el.id !== id)),
        EDIT_PROJECT: (state, project) =>
            (state.data = state.data.map((el) =>
                el.id === project.id ? project : el
            )),
    },
    actions: {
        async getAllProjects({ commit,dispatch }) {
            try {
                const userRes = await this.$axios.get("/user");
                const { data } = await this.$axios.get(
                    `/works/${userRes.data.user.id}`
                );
                commit("GET_PROJECTS", data);
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async addNewProject({ commit, dispatch }, project) {
            try {
                const formData = new FormData();
                await Object.keys(project).forEach((item) => {
                    formData.append(item, project[item]);
                });
                const { data } = await this.$axios.post("/works", formData);
                console.log(data);
                commit("ADD_PROJECT", data);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Добавлен новый проект",
                        type: "Succsess",
                    },
                    { root: true }
                );

            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async deleteProject({ commit, dispatch }, id) {
            try {
                const { data } = await this.$axios.delete(`/works/${id}`);
                commit("DELETE_PROJECT", id);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Проект успешно удален",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async editProject({ commit, dispatch }, project) {
            try {
                const formData = new FormData();
                await Object.keys(project).forEach((item) => {
                    formData.append(item, project[item]);
                });
                const { data } = await this.$axios.post(
                    `/works/${project.id}`,
                    formData
                );
                commit("EDIT_PROJECT", data.work);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Проект успешно изменен",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
    },
};
