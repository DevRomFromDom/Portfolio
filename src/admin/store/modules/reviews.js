export default {
    namespaced: true,
    state: {
        data: [],
    },
    mutations: {
        ADD_REVIEW: (state, data) => state.data.push(data),
        GET_REVIEWS: (state, data) => (state.data = data),
        DELETE_REVIEW: (state, id) =>
            (state.data = state.data.filter((el) => el.id !== id)),
        EDIT_REVIEW: (state, review) =>
            (state.data = state.data.map((el) =>
                el.id === review.id ? review : el
            )),
    },
    actions: {
        async getAllReviews({ commit,dispatch }) {
            try {
                const userRes = await this.$axios.get("/user");
                const { data } = await this.$axios.get(
                    `/reviews/${userRes.data.user.id}`
                );
                commit("GET_REVIEWS", data);
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async addNewReview({ commit, dispatch }, review) {
            try {
                const formData = new FormData();
                await Object.keys(review).forEach((item) => {
                    formData.append(item, review[item]);
                });
                const { data } = await this.$axios.post("/reviews", formData);
                commit("ADD_REVIEW", data);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Добавлен новый проект",
                        type: "Succsess",
                    },
                    { root: true }
                );

            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async deleteReview({ commit, dispatch }, id) {
            try {
                const { data } = await this.$axios.delete(`/reviews/${id}`);
                commit("DELETE_REVIEW", id);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Проект успешно удален",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
        async editReview({ commit, dispatch }, review) {
            try {
                const formData = new FormData();
                await Object.keys(review).forEach((item) => {
                    formData.append(item, review[item]);
                });
                const { data } = await this.$axios.post(
                    `/reviews/${review.id}`,
                    formData
                );
                commit("EDIT_REVIEW", data.review);
                dispatch(
                    "tooltips/show",
                    {
                        text: "Проект успешно изменен",
                        type: "Succsess",
                    },
                    { root: true }
                );
            } catch (error) {
                dispatch(
                    "tooltips/show",
                    {
                        text: error.response.data.error,
                        type: "error",
                    },
                    { root: true }
                );
            }
        },
    },
};
