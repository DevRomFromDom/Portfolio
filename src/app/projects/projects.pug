section.projects
  .projects__container 
    .projects-title Мои работы
    #projects-container
    