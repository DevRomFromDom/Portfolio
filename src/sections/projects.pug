section.projects
  .projects__container 
    .projects-title Мои работы
    .projects__content 
      .projects__content-slider
        .slider__preview
          +image('content/preview-1.jpg',"preview")
          .slider__miniature
            - for (let i = 1;i<=4;i++)
              .miniature__item(data-slide=`${2}`)
                +image(`content/preview-${i}.jpg`,"preview")
        .nav__container
          nav.slider__nav
            .nav-button
              .nav-arrow.next
                +icon({"name":'arrowNext'},"slider-next")
            .nav-button
              .nav-arrow.passed
                +icon({"name":'arrowPassed'},"slider-passed")
      .projects__content-info
        ul.projects-info__tags
          - var projTags = ["HTML", "CSS", "Javascript"];
          each item in projTags
            li.project-tag__item 
              .tag-text #{item}
        .projects-info__text 
          .project__text-title Сайт школы #[br] образования
          .project__text-content Этот парень проходил обучение #[br]веб-разработке не где-то, а в LoftSchool!#[br] 2 месяца только самых тяжелых #[br]испытаний и бессонных ночей!
        .projects-info__link
          .project__link-icon 
            +icon({"name":"link","height":"22.63px","width":"22.63px"},"project-link-svg")
          .project__link-text Посмотреть сайт
