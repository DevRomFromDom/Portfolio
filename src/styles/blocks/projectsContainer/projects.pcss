.projects_container {
    position: absolute;
    width: 1469px;
    height: 681px;
    left: 158px;
    top: 2997px;
}
.projects_title {
    /* Мои работы */

    position: absolute;
    width: 232px;
    height: 49px;
    left: 2px;
    top: 0px;

    /* Desktop/h2 */

    font-family: Open Sans;
    font-style: normal;
    font-weight: bold;
    font-size: 36px;
    line-height: 49px;
    /* identical to box height */

    display: flex;
    align-items: center;

    /* dark */

    color: #222427;
}

.projects_slider {
    position: absolute;
    width: 1028px;
    height: 549px;
    left: 0px;
    top: 132px;
}
.projects_print_two {
    /* 2 */

    position: absolute;
    width: 235px;
    height: 311px;
    left: 915px;
    top: 209px;

    font-family: Open Sans;
    font-style: normal;
    font-weight: 800;
    font-size: 400px;
    line-height: 48px;
    /* or 12% */

    display: flex;
    align-items: center;

    /* light grey 2 */

    color: #e6e6e6;
}

.projects_info {
    position: absolute;
    width: 338px;
    height: 391px;
    left: 1131px;
    top: 135px;
}
.projects_info_tags {
    position: absolute;
    display: flex;
    justify-content: space-between;
    width: 288px;
    height: 48px;
    left: 3px;
    top: 0px;
}
.tag {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 14px 22px;
    width: auto;
    height: 48px;
    /* tag */
    background: #f3f6f9;
    border-radius: 40px;
}

.projects_info_title {
    /* Сайт школы образования */

    position: absolute;
    width: 207px;
    height: 82px;
    left: 3px;
    top: 83px;

    /* Desktop/h3 */

    font-family: Open Sans;
    font-style: normal;
    font-weight: bold;
    font-size: 30px;
    line-height: 41px;
    display: flex;
    align-items: center;

    /* dark */

    color: #222427;
}

.projects_info_text {
    /* Этот парень проходил обучение веб-разработке не где-то, а в Lof */

    position: absolute;
    width: 336px;
    height: 120px;
    left: 2px;
    top: 200px;

    /* Desktop/Body 1 SemiBold */

    font-family: Open Sans;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 30px;
    /* or 187% */

    display: flex;
    align-items: center;

    /* grey 70% */

    color: rgba(51, 60, 73, 0.7);
}

.projects_info_link {
    position: absolute;
    width: 200px;
    height: 26px;
    left: 1131px;
    top: 500px;
    border-color: black;
}
.projects_info_link_icon {
    height: 10px;
    width: 22px;
    background: svg-load(
        "link.svg",
        fill=#586ED3,
        width=22PX,
        height=10PX
    );
    cursor: pointer;
    border-radius: 0px;
    transform: rotate(-45deg);
}
.projects_info_link_text {
    /* Посмотреть сайт */

    position: absolute;
    width: 161px;
    height: 25px;
    left: 1328px;
    top: 3498px;

    font-family: Open Sans;
    font-style: normal;
    font-weight: bold;
    font-size: 18px;
    line-height: 25px;
    /* identical to box height */

    display: flex;
    align-items: center;

    /* link */

    color: #586ed3;
}
