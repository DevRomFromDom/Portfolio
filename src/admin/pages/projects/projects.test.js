import projects from "./projects.vue";
import Vuex from "vuex";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import { state, mutations } from "../../store/modules/projects";

const localVue = createLocalVue();
localVue.use(Vuex);

let store, actions;

beforeEach(() => {
    actions = {
        getProjectsAction: jest.fn(),
        removeProjectAction: jest.fn(),
    };

    store = new Vuex.Store({
        namespaced: true,
        state: { ...state },
        mutations: { ...mutations },
        actions,
    });
});
afterEach(() => {
    store = {};
});

it("Component has been mounted", async () => {
    const wrapper = shallowMount(
        projects,
        { store, localVue },
        {
            propsData: {
                projectModalIsAcrive: false,
                modalTitle: "Добавление работы",
                editProjectData: {},
            },
        }
    );
    expect(wrapper.element).toMatchSnapshot();
});
