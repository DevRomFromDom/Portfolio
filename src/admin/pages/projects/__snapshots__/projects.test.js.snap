// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component has been mounted 1`] = `
<div
  class="page-content"
>
  <div
    class="header"
  >
    <div
      class="title"
    >
      Блок «Работы»
    </div>
  </div>
   
  <div
    class="projects__content-container"
  >
    <!---->
     
    <div
      class="create-project"
    >
      <square-btn-stub
        class="create-button"
        title="Добавить работу"
        type="square"
      />
    </div>
     
  </div>
</div>
`;
